Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp

```diff
@@ -309,6 +315,15 @@ void BitmapTexture::copyFromExternalTexture(GLuint sourceTextureID, const IntRec
 {
     RELEASE_ASSERT(sourceOffset.width() + targetRect.width() <= m_size.width());
     RELEASE_ASSERT(sourceOffset.height() + targetRect.height() <= m_size.height());
+#if USE(SKIA)
+    RELEASE_ASSERT(m_colorConvertFlags.isEmpty());
```

Comment:
Same as the updateContents case, changing the format is not always wrong. It's fine if we update the whole texture. What we need to assert is that we're not changing the format with partial updates. If we don't allow format changes when they cover the whole texture, then a texture that contains one format can never be reused from the pool to hold the other format.